pgjdbc/src/main/java/org/postgresql/Driver.java

```diff
@@ -288,14 +262,65 @@ private Properties loadDefaultProperties() throws IOException {
               "Your security policy has prevented the connection from being attempted.  You probably need to grant the connect java.net.SocketPermission to the database server host and port that you wish to connect to."),
           PSQLState.UNEXPECTED_ERROR, ace);
     } catch (Exception ex2) {
-      logger.debug("Unexpected connection error:", ex2);
+      LOGGER.log(Level.SEVERE, "Unexpected connection error: ", ex2);
       throw new PSQLException(
           GT.tr(
               "Something unusual has occurred to cause the driver to fail. Please report this exception."),
           PSQLState.UNEXPECTED_ERROR, ex2);
     }
   }
 
+  // Used to check if the handler file is the same
+  private static String loggerHandlerFile;
+
+  /**
+   * Setup java.util.logging.Logger using connection properties.
+   * <p>
+   * See {@link PGProperty#LOGGER_FILE} and {@link PGProperty#LOGGER_FILE}
+   *
+   * @param props Connection Properties
+   */
+  private void setupLoggerFromProperties(final Properties props) {
+    final String driverLogLevel = PGProperty.LOGGER_LEVEL.get(props);
+    if (driverLogLevel == null) {
+      return; // Don't mess with Logger if not set
+    }
+    if ("OFF".equalsIgnoreCase(driverLogLevel)) {
+      PARENT_LOGGER.setLevel(Level.OFF);
+      return; // Don't mess with Logger if set to OFF
+    } else if ("DEBUG".equalsIgnoreCase(driverLogLevel)) {
+      PARENT_LOGGER.setLevel(Level.FINE);
+    } else if ("TRACE".equalsIgnoreCase(driverLogLevel)) {
+      PARENT_LOGGER.setLevel(Level.FINEST);
+    }
+
+    final String driverLogFile = PGProperty.LOGGER_FILE.get(props);
+    if (driverLogFile != null && driverLogFile.equals(loggerHandlerFile)) {
+      return; // Same file output, do nothing.
+    }
+
+    for (java.util.logging.Handler handlers : PARENT_LOGGER.getHandlers()) {
+      // Remove previously set Handlers
+      handlers.close();
+      PARENT_LOGGER.removeHandler(handlers);
+      loggerHandlerFile = null;
+    }
+
+    java.util.logging.Handler handler = new java.util.logging.ConsoleHandler();
+    if (driverLogFile != null) {
+      try {
+        handler = new java.util.logging.FileHandler(driverLogFile);
+        loggerHandlerFile = driverLogFile;
+      } catch (Exception ex) {
+        System.err.println("Cannot enable FileHandler, fallback to ConsoleHandler.");
+      }
+    }
+
+    handler.setFormatter(new java.util.logging.SimpleFormatter());
+    PARENT_LOGGER.setUseParentHandlers(false);
```

Comment:
Technically parent handler are defined for the root Logger (.) And by default it is a console handler at level info, since we are programatically defining a handler, that explicitly means don't use console, use only what I want, (file or console).

But if you prefer I can remove that line.